### PR TITLE
update pid loop

### PR DESCRIPTION
**Pull request by MichaelMohn**
# Quality Assurance Checklist 
To make reviews more efficient, please make sure the software feature meets the following standards and check everything off that meets the quality check. Once everything has been checked, the assigned reviewers will begin the review process. _Edit this description to check off the list._

There are exceptions with all guidelines. As long as your decisions are justified, then you are good! Contact the reviewers or the leads about any exceptions. 

## Requirements 
- [x] Followed Coding Style Guide 
- [x] Code Build checks pass 
- [x] No merge conflicts 
- [ ] Software feature has test associated with it. 
- [ ] Software feature has documentation for it updated.
- [x] If applicable, does software feature have simulation test associated with it (.json & out.json file)
- [ ] Test provides useful information and uses relevant data to accurately represent BPS 
    - NOTE: If test file already exists, use that one 
- [ ] 2 Members of the team have already reviewed my PR 
    - NOTE: Approver will not look at PR if 2 other members haven't reviewed first 
- [x] Did you Have Fun?

## Things to Consider 
- [ ] Even if the above are checked, is this the best way of writing my code? 
    - It's possible to write code that works, but are there ways to make code more efficient?

### CODE REVIEW

File: Apps/Src/Temperature.c [hunk 1]
```diff
@@ -420,22 +419,42 @@ int32_t Temperature_GetMaxTemperature(void) {
...
+			ErrorSum = I_MAX_ACCUM;
+		}else{
+			ErrorSum = -I_MAX_ACCUM;
+		}
+	}
+
+	//5 is the estimated "hold output"
+	int32_t p_Output = ((-Error)/PROPORTION*1000) + 5; //Scale P-output to 0-8, Floor divide error by 1000
+
+	//I output could totally fuck things up (and probably will on the first test), so disable it and make sure p is good first
```
Comment (connorl309) [inline Apps/Src/Temperature.c:446]: remove the funny word from ur comment

File: Apps/Src/Temperature.c [hunk 2]
```diff
@@ -420,22 +419,42 @@ int32_t Temperature_GetMaxTemperature(void) {
...
+	//Only read error sum in range
+	if(abs(Error) < I_ZONE){
+		ErrorSum += Error;
+	}else{
+		ErrorSum = 0;
+	}
+
+	//Cap error sum at 500 read degrees
+	if(abs(ErrorSum) > I_MAX_ACCUM){
+		if(ErrorSum > 0){
```
Comment (connorl309) [inline Apps/Src/Temperature.c:436]: just do ErrorSum = abs(I_MAX_ACCUM)?

**Review by connorl309 — COMMENTED** [review]
i dont know a whole lot about PID but a couple changes to make. overall structure looks good

File: Apps/Src/Temperature.c (hunk 2 above)
Comment (ray4477) [reply Apps/Src/Temperature.c:436]: I don't think that would work because you're supposed to see if error goes over the max accum

File: Apps/Src/Temperature.c (hunk 1 above)
Comment (ray4477) [reply Apps/Src/Temperature.c:446]: nah that can be left there fr

File: Apps/Src/Temperature.c [hunk 3]
```diff
@@ -420,22 +419,42 @@ int32_t Temperature_GetMaxTemperature(void) {
...
+
+	//5 is the estimated "hold output"
+	int32_t p_Output = ((-Error)/PROPORTION*1000) + 5; //Scale P-output to 0-8, Floor divide error by 1000
+
+	//I output could totally fuck things up (and probably will on the first test), so disable it and make sure p is good first
+	//Keep I gains low or you'll get weird oscillation. abs(I output) should not currently exceed 2
+	int32_t i_Output = ((-ErrorSum)/INTEGRAL*1000);
+
+	//Don't use D output
+	int32_t output = p_Output + i_Output;
```
Comment (ray4477) [inline Apps/Src/Temperature.c:451]: can be uint8_t

File: Apps/Src/Temperature.c (hunk 2 above)
Comment (MichaelMohn) [reply Apps/Src/Temperature.c:436]: nah i did that because error sum needs to be able to be negative

**Review by Champers5000 — APPROVED** [review]
PID loop looks ok, has checks to ensure things don't go too horribly wrong

**Review by ray4477 — CHANGES_REQUESTED** [review]
Change output to uint8_t if it only returns values from 0-8. I'm pretty sure your pid code works but instead of doing `(-ErrorSum)/INTEGRAL*1000` do `(-ErrorSum)*1000/INTEGRAL`. The otherway kind of gets rid of precison I'm p sure.

File: Apps/Src/Temperature.c (hunk 1 above)
Comment (manthanand) [reply Apps/Src/Temperature.c:446]: lets keep it pg for now. maybe fudge

File: Apps/Src/Temperature.c [hunk 4]
```diff
@@ -420,22 +419,42 @@ int32_t Temperature_GetMaxTemperature(void) {
  * @brief Gives fan speed based on Average temperature of pack and past error values
  * @param InputTemp - current temperature
  * @param DesiredTemp - desired temperature
- * @return FanSpeed: 0-4000 PWM
+ * @return FanSpeed: 0-8
  */
 int32_t Temperature_PID_Output(int32_t InputTemp, int32_t DesiredTemp) {
```
Comment (manthanand) [inline Apps/Src/Temperature.c:424]: Why are you returning an int32 if the range of values is 0-8? Also, Idk what exactly your code does, and I'm kinda getting lost in the sauce with your algorithm (and the negative numbers). You should update the documentation for this

File: Apps/Src/Temperature.c [hunk 5]
```diff
@@ -420,22 +419,42 @@ int32_t Temperature_GetMaxTemperature(void) {
...
+	if(abs(ErrorSum) > I_MAX_ACCUM){
+		if(ErrorSum > 0){
+			ErrorSum = I_MAX_ACCUM;
+		}else{
+			ErrorSum = -I_MAX_ACCUM;
+		}
+	}
+
+	//5 is the estimated "hold output"
+	int32_t p_Output = ((-Error)/PROPORTION*1000) + 5; //Scale P-output to 0-8, Floor divide error by 1000
```
Comment (manthanand) [inline Apps/Src/Temperature.c:444]: should make `#define` for 1000 and 5 and write comments next to them saying what they are for. Same for the other defines that you added to the code.

File: Apps/Src/Temperature.c [hunk 6]
```diff
@@ -43,16 +43,15 @@ static uint8_t currentChannel = 0;
...
-#define PROPORTION -1234
-#define INTEGRAL 4321
-#define DERIVATIVE -9876
-#define MAX_FAN_SPEED 4000
-#define DIVISOR 25000
+#define PROPORTION 2
+#define INTEGRAL 250
+#define I_ZONE 3000
+#define I_MAX_ACCUM 500000
+#define MAX_FAN_SPEED 8
```
Comment (manthanand) [inline Apps/Src/Temperature.c:50]: There is already a TOPSPEED define in Fans.h. You can either rename that to this or just use that. Having 2 defines for the same purpose is not a good idea.

File: Apps/Src/Temperature.c (hunk 2 above)
Comment (tiandahuang) [reply Apps/Src/Temperature.c:436]: could replace all these if elses with ternaries

File: Apps/Src/Temperature.c (hunk 6 above)
Comment (tiandahuang) [reply Apps/Src/Temperature.c:50]: ```static const uint32_t``` is a good alternative for local stuff, but these tunable parameters really should be in the header file

also name your ```#define``` more specific, i.e. ```TEMPERATURE_PID_KP``` or something, tbh make em long so we don't get any conflicts and so their purposes are very crystal clear

File: Apps/Src/Temperature.c [hunk 7]
```diff
@@ -420,22 +419,42 @@ int32_t Temperature_GetMaxTemperature(void) {
...
-    if (((PROPORTION*(Error) + INTEGRAL*(ErrorSum) + DERIVATIVE*(Rate))/DIVISOR) > MAX_FAN_SPEED) {
-        return MAX_FAN_SPEED;
-    }
-    if (((PROPORTION*(Error) + INTEGRAL*(ErrorSum) + DERIVATIVE*(Rate))/DIVISOR) <= 0) {
-        return 0;
-    }
-    return (PROPORTION*(Error) + INTEGRAL*(ErrorSum) + DERIVATIVE*(Rate))/DIVISOR;
+	
+	//Only read error sum in range
+	if(abs(Error) < I_ZONE){
```
Comment (tiandahuang) [inline Apps/Src/Temperature.c:428]: I think ternaries are faster

File: Apps/Src/Temperature.c [hunk 8]
```diff
@@ -420,22 +419,42 @@ int32_t Temperature_GetMaxTemperature(void) {
...
+	//Don't use D output
+	int32_t output = p_Output + i_Output;
+
+	if(output > MAX_FAN_SPEED){
+		return MAX_FAN_SPEED;
+	}else if(output < 0){
+		return 0;
+	}else{
+		return output;
+	}
```
[the remark this answers is not in the document]
Comment (tiandahuang) [reply Apps/Src/Temperature.c:459]: it would help to use spaces between things like ```if (``` and ```} else {```

**Review by tiandahuang — CHANGES_REQUESTED** [review]
please please read the style guide

File: Apps/Src/Temperature.c (hunk 1 above)
Comment (MichaelMohn) [reply Apps/Src/Temperature.c:446]: fixed

File: Apps/Src/Temperature.c (hunk 2 above)
Comment (MichaelMohn) [reply Apps/Src/Temperature.c:436]: fixed

Comment (MichaelMohn) [reply Apps/Src/Temperature.c:436]: fixed

**Comment by MichaelMohn** [comment]
> Change output to uint8_t if it only returns values from 0-8. I'm pretty sure your pid code works but instead of doing `(-ErrorSum)/INTEGRAL*1000` do `(-ErrorSum)*1000/INTEGRAL`. The otherway kind of gets rid of precison I'm p sure.

Nah I might not understand this part of C perfectly... but at the moment ErrorSum is capped at 500,000 and integral is 250. So the max output is 2. If I place the 1000 in the numerator I would get a massively incorrect value right?

File: Apps/Src/Temperature.c [hunk 9]
```diff
@@ -420,22 +419,42 @@ int32_t Temperature_GetMaxTemperature(void) {
...
+			ErrorSum = -I_MAX_ACCUM;
+		}
+	}
+
+	//5 is the estimated "hold output"
+	int32_t p_Output = ((-Error)/PROPORTION*1000) + 5; //Scale P-output to 0-8, Floor divide error by 1000
+
+	//I output could totally fuck things up (and probably will on the first test), so disable it and make sure p is good first
+	//Keep I gains low or you'll get weird oscillation. abs(I output) should not currently exceed 2
+	int32_t i_Output = ((-ErrorSum)/INTEGRAL*1000);
```
Comment (manthanand) [inline Apps/Src/Temperature.c:448]: For this code, if you are trying to multiply the fraction by 1000, then you need to have 1000 in the numerator. Otherwise, the first operation will round to 0 and you will always get 0 as the output (even if the value could be 1 or otherwise).

Comment (manthanand) [reply Apps/Src/Temperature.c:448]: If the denominator is to be multiplied by 1000, then change the parentheses to make it look like that.

Comment (manthanand) [reply Apps/Src/Temperature.c:448]: `-ErrorSum/(INTEGRAL*1000)`

Comment (MichaelMohn) [reply Apps/Src/Temperature.c:448]: fixed

File: Apps/Src/Temperature.c (hunk 6 above)
Comment (MichaelMohn) [reply Apps/Src/Temperature.c:50]: fixed

File: Apps/Src/Temperature.c (hunk 5 above)
Comment (MichaelMohn) [reply Apps/Src/Temperature.c:444]: fixed

File: Apps/Src/Temperature.c (hunk 8 above)
[the remark this answers is not in the document]
Comment (MichaelMohn) [reply Apps/Src/Temperature.c:459]: fixed

File: Apps/Src/Temperature.c (hunk 4 above)
Comment (MichaelMohn) [reply Apps/Src/Temperature.c:424]: fixed

Comment (manthanand) [reply Apps/Src/Temperature.c:418]: this isn't fixed

File: Apps/Src/Temperature.c [hunk 10]
```diff
@@ -420,22 +413,26 @@ int32_t Temperature_GetMaxTemperature(void) {
...
 int32_t Temperature_PID_Output(int32_t InputTemp, int32_t DesiredTemp) {
     Error = DesiredTemp - InputTemp;
-    ErrorSum = ErrorSum + Error;
+	
+    //Only read error sum in range
+    ErrorSum += abs(Error) < TEMPERATURE_PID_I_ZONE ? Error : 0;
 
-    if (PreviousError == 0) {PreviousError = Error;} //init previous val first time
+	//Cap error sum at 500 read degrees
+    ErrorSum = abs(ErrorSum) > TEMPERATURE_PID_I_MAX_ACCUM ? ErrorSum > 0 ? TEMPERATURE_PID_I_MAX_ACCUM : -TEMPERATURE_PID_I_MAX_ACCUM : ErrorSum;
```
Comment (manthanand) [inline Apps/Src/Temperature.c:425]: add some parentheses here to make the ternaries easier to read. You can also write it in different lines so it is easier to understand.
```
ErrorSum = (abs(ErrorSum) > TEMPERATURE_PID_I_MAX_ACCUM) ? 
    ((ErrorSum > 0) ? TEMPERATURE_PID_I_MAX_ACCUM : -TEMPERATURE_PID_I_MAX_ACCUM) : ErrorSum;
```

File: Apps/Src/Temperature.c [hunk 11]
```diff
@@ -420,22 +413,26 @@ int32_t Temperature_GetMaxTemperature(void) {
...
 
-    Rate = Error - PreviousError;
-    PreviousError = Error;     //updates previous err value
+	//5 is the estimated "hold output"
```
Comment (manthanand) [inline Apps/Src/Temperature.c:427]: you should still have the 5 as a define at the top of the file

File: Apps/Src/Temperature.c [hunk 12]
```diff
@@ -420,22 +413,26 @@ int32_t Temperature_GetMaxTemperature(void) {
...
-        return 0;
-    }
-    return (PROPORTION*(Error) + INTEGRAL*(ErrorSum) + DERIVATIVE*(Rate))/DIVISOR;
+	//I output could totally fudge things up (and probably will on the first test), so disable it and make sure p is good first
+	//Keep I gains low or you'll get weird oscillation. abs(I output) should not currently exceed 2
+	int32_t i_Output = ((-ErrorSum)/(TEMPERATURE_PID_INTEGRAL*TEMPERATURE_PID_MILICELCIUS_CONVERT));
+
+	//Don't use D output
+	int32_t output = p_Output + i_Output;
+    output = output > TOPSPEED ? TOPSPEED : output < 0 ? 0 : output;
```
Comment (manthanand) [inline Apps/Src/Temperature.c:436]: same as other comment with ternary

File: Apps/Src/Temperature.c (hunk 10 above)
Comment (MichaelMohn) [reply Apps/Src/Temperature.c:425]: fixed

File: Apps/Src/Temperature.c (hunk 11 above)
Comment (MichaelMohn) [reply Apps/Src/Temperature.c:427]: fixed

File: Apps/Src/Temperature.c (hunk 12 above)
Comment (MichaelMohn) [reply Apps/Src/Temperature.c:436]: fixed

**Comment by ray4477** [comment]
> > Change output to uint8_t if it only returns values from 0-8. I'm pretty sure your pid code works but instead of doing `(-ErrorSum)/INTEGRAL*1000` do `(-ErrorSum)*1000/INTEGRAL`. The otherway kind of gets rid of precison I'm p sure.
> 
> Nah I might not understand this part of C perfectly... but at the moment ErrorSum is capped at 500,000 and integral is 250. So the max output is 2. If I place the 1000 in the numerator I would get a massively incorrect value right?

It should be equivalent mathematical statements. The problem with the before is if ErrorSum is say less than 250 than if it is divided by INTEGRAL (250) it will become zero because int division rounds down. If you do the other way then it will return fine. Here is an example below
image

**Comment by MichaelMohn** [comment]
> > > Change output to uint8_t if it only returns values from 0-8. I'm pretty sure your pid code works but instead of doing `(-ErrorSum)/INTEGRAL*1000` do `(-ErrorSum)*1000/INTEGRAL`. The otherway kind of gets rid of precison I'm p sure.
> > 
> > 
> > Nah I might not understand this part of C perfectly... but at the moment ErrorSum is capped at 500,000 and integral is 250. So the max output is 2. If I place the 1000 in the numerator I would get a massively incorrect value right?
> 
> It should be equivalent mathematical statements. The problem with the before is if ErrorSum is say less than 250 than if it is divided by INTEGRAL (250) it will become zero because int division rounds down. If you do the other way then it will return fine. Here is an example below image

This has been fixed I believe.

**Review by tiandahuang — APPROVED** [review]

File: Apps/Src/Temperature.c [hunk 13]
```diff
@@ -418,24 +411,31 @@ int32_t Temperature_GetMaxTemperature(void) {
...
-    if (PreviousError == 0) {PreviousError = Error;} //init previous val first time
+    //Cap error sum at 500 read degrees
+    ErrorSum = (abs(ErrorSum) < TEMPERATURE_PID_I_MAX_ACCUM) ? ErrorSum :
+        (ErrorSum > 0 ? TEMPERATURE_PID_I_MAX_ACCUM : -TEMPERATURE_PID_I_MAX_ACCUM);
 
-    Rate = Error - PreviousError;
-    PreviousError = Error;     //updates previous err value
+    //5 is the estimated "hold output"
+    //Scale P-output to 0-8, Floor divide error by 1000
+    int8_t p_Output = (int8_t)((-Error)/(TEMPERATURE_PID_PROPORTIONAL*TEMPERATURE_PID_MILICELCIUS_CONVERT)) + TEMPERATURE_HOLD_FAN_SPEED;
```
[the remark this answers is not in the document]
Comment (tiandahuang) [reply Apps/Src/Temperature.c:430]: in addition you could add divide with rounding instead of divide with truncating:
https://stackoverflow.com/questions/2422712/rounding-integer-division-instead-of-truncating

File: Apps/Src/Temperature.c [hunk 14]
```diff
@@ -418,24 +411,31 @@ int32_t Temperature_GetMaxTemperature(void) {
...
- * @return FanSpeed: 0-4000 PWM
+ * @return FanSpeed: 0-8
  */
-int32_t Temperature_PID_Output(int32_t InputTemp, int32_t DesiredTemp) {
+int8_t Temperature_PID_Output(int32_t InputTemp, int32_t DesiredTemp) {
     Error = DesiredTemp - InputTemp;
-    ErrorSum = ErrorSum + Error;
+	
+    //Only read error sum in range
+    ErrorSum += abs(Error) < TEMPERATURE_PID_I_ZONE ? Error : 0;
```
Comment (ray4477) [inline Apps/Src/Temperature.c:422]: why is this here, reading errors only in range of 3 C? Temperature_PID_I_ZONE is only 3000 which is 3 celsius. idk but feels low to me

File: Apps/Src/Temperature.c [hunk 15]
```diff
@@ -418,24 +411,31 @@ int32_t Temperature_GetMaxTemperature(void) {
 
 /**
  * @brief Gives fan speed based on Average temperature of pack and past error values
- * @param InputTemp - current temperature
+ * @param InputTemp - current temperature. Must not exceed 284 or -284 Celcius given that expected temp is 38 celcius.
```
[the remark this answers is not in the document]
Comment (MichaelMohn) [reply Apps/Src/Temperature.c:414]: yeah i figured that if the car has reached 10 degrees below absolute zero that there would be more pressing issues at hand than the fans

File: Apps/Src/Temperature.c (hunk 14 above)
Comment (MichaelMohn) [reply Apps/Src/Temperature.c:422]: 3 was a guess, i actually just increased it to 5. Regardless this number is very low because you don't want to use the "i-part" of the pid unless you are very close to the target. 

Error, instead of errorsum, is used by the "p-part" and is what is read and updated regardless of how close to the target we are.

File: Apps/Src/Temperature.c [hunk 16]
```diff
@@ -418,24 +411,31 @@ int32_t Temperature_GetMaxTemperature(void) {
...
-        return 0;
-    }
-    return (PROPORTION*(Error) + INTEGRAL*(ErrorSum) + DERIVATIVE*(Rate))/DIVISOR;
+    //I output could totally fudge things up (and probably will on the first test), so disable it and make sure p is good first
+    //Keep I gains low or you'll get weird oscillation. abs(I output) should not currently exceed 2 
+    int8_t i_Output = (int8_t)((-ErrorSum)/(TEMPERATURE_PID_INTEGRAL*TEMPERATURE_PID_MILICELCIUS_CONVERT));
+    
+    //Don't use D output
+    int8_t output = p_Output + i_Output;
+    output = (output > TOPSPEED) ? TOPSPEED :
```
[the remark this answers is not in the document]
Comment (MichaelMohn) [reply Apps/Src/Temperature.c:438]: good idea

File: Apps/Src/Temperature.c [hunk 17]
```diff
@@ -418,24 +411,31 @@ int32_t Temperature_GetMaxTemperature(void) {
...
 /**
  * @brief Gives fan speed based on Average temperature of pack and past error values
- * @param InputTemp - current temperature
+ * @param InputTemp - current temperature. Must not exceed 284 or -284 Celcius given that expected temp is 38 celcius.
  * @param DesiredTemp - desired temperature
- * @return FanSpeed: 0-4000 PWM
+ * @return FanSpeed: 0-8
  */
-int32_t Temperature_PID_Output(int32_t InputTemp, int32_t DesiredTemp) {
+int8_t Temperature_PID_Output(int32_t InputTemp, int32_t DesiredTemp) {
```
[the remark this answers is not in the document]
Comment (MichaelMohn) [reply Apps/Src/Temperature.c:418]: oh yeah

File: Apps/Src/Temperature.c (hunk 14 above)
Comment (MichaelMohn) [reply Apps/Src/Temperature.c:422]: jk going back to three

**Review by ray4477 — APPROVED** [review]

File: Apps/Src/Temperature.c (hunk 13 above)
[the remark this answers is not in the document]
Comment (MichaelMohn) [reply Apps/Src/Temperature.c:430]: ok I'm dividing my millicelcius at the end for output

[the remark this answers is not in the document]
Comment (MichaelMohn) [reply Apps/Src/Temperature.c:430]: fixed

File: Apps/Src/Temperature.c (hunk 17 above)
[the remark this answers is not in the document]
Comment (MichaelMohn) [reply Apps/Src/Temperature.c:418]: fixed

File: Apps/Src/Temperature.c (hunk 15 above)
[the remark this answers is not in the document]
Comment (manthanand) [reply Apps/Src/Temperature.c:414]: you should provide error checking if we exceed these bounds. What happens if the code accidentally puts in those values when the actual temperature is fine or within bounds?

File: Apps/Inc/Temperature.h [hunk 18]
```diff
@@ -130,5 +138,5 @@ int32_t Temperature_GetMaxTemperature(void);
  * @param DesiredTemp - desired temperature
  * @return FanSpeed: 0-4000 PWM
  */
-int32_t Temperature_PID_Output(int32_t InputTemp, int32_t DesiredTemp);
+uint8_t Temperature_PID_Output(int32_t InputTemp, int32_t DesiredTemp);
```
Comment (manthanand) [inline Apps/Inc/Temperature.h:141]: the inputs to this should also be all positive. Shouldn't they be uint32_t?

File: Apps/Src/Temperature.c (hunk 15 above)
[the remark this answers is not in the document]
Comment (MichaelMohn) [reply Apps/Src/Temperature.c:414]: fixed

File: Apps/Inc/Temperature.h (hunk 18 above)
Comment (MichaelMohn) [reply Apps/Inc/Temperature.h:141]: fixed

File: Tests/Test_Fault_Lights.c [hunk 19]
```diff
@@ -0,0 +1,67 @@
...
+    Fault_OTEMP   = 0x004,
+    Fault_OCURR   = 0x008,
+    Fault_OW      = 0x010,
+    Fault_HANDLER = 0x020,
+    Fault_OS      = 0x040,
+    Fault_WDOG    = 0x080,
+    Fault_CRC     = 0x100,
+    Fault_ESTOP   = 0x200,
+    Fault_Max
+}Fault_Set;
```
Comment (Champers5000) [inline Tests/Test_Fault_Lights.c:24]: Wasn't this defined elsewhere already?

**Review by Champers5000 — DISMISSED** [review]

File: Apps/Inc/Temperature.h [hunk 20]
```diff
@@ -130,5 +139,5 @@ int32_t Temperature_GetMaxTemperature(void);
  * @param DesiredTemp - desired temperature
  * @return FanSpeed: 0-4000 PWM
```
Comment (manthanand) [inline Apps/Inc/Temperature.h:140]: change comment here.

File: Apps/Src/Temperature.c [hunk 21]
```diff
@@ -418,24 +411,34 @@ int32_t Temperature_GetMaxTemperature(void) {
 
 /**
  * @brief Gives fan speed based on Average temperature of pack and past error values
- * @param InputTemp - current temperature
+ * @param InputTemp - current temperature. Must not exceed 293 or -294 Celcius given that expected temp is 38 celcius.
```
Comment (manthanand) [inline Apps/Src/Temperature.c:414]: Write what happens if temperature does exceed these bounds.

File: Tests/Test_Fault_Lights.c (hunk 19 above)
Comment (MichaelMohn) [reply Tests/Test_Fault_Lights.c]: i didnt mean to push this file oops

File: Apps/Inc/Temperature.h (hunk 20 above)
Comment (MichaelMohn) [reply Apps/Inc/Temperature.h:140]: fixed

**Review by manthanand — APPROVED** [review]
